### PR TITLE
aii schema: add protected option

### PR DESCRIPTION
**Pull request by kwaegema**
added value for https://github.com/quattor/aii/pull/172

### CODE REVIEW

**Comment by jrha** [comment]
No validation? :disappointed:
No annotation? :cry:
What are expected/example values?

**Comment by kwaegema** [comment]
@jrha added a docstring.
We use it as folows:
'/system/aii/protected' = value('/system/quattorid');
Then this is the result:

> [root@gastly ~]# aii-rci test2200.shuppet.os
[INFO] remove step for reinstall option.
[ERROR] Host test2200.shuppet.os is protected. Specify --protectid lsimngar-swablu-1463997392 to overwrite.
[ERROR] No nodes left to process after checking for protected hosts

File: quattor/types/aii.pan [hunk 1]
```diff
@@ -30,9 +30,17 @@ type structure_aii_hooklist = {
     "status" ? structure_aii_hook[]
 };
 
+@documentation{
+    AII structure which defines the osinstall, nbp,
+    dhcp config and hooks.
+    The "protected" value can be used to protect the install and removal.
```
Comment (stdweird) [inline quattor/types/aii.pan]: make annotation above the attribute like
```pan
@{...}
"protected" ? string
```

**Comment by stdweird** [comment]
@ned21 any remarks?

File: quattor/types/aii.pan [hunk 2]
```diff
@@ -30,9 +30,17 @@ type structure_aii_hooklist = {
...
+    By setting this value (eg. to the latest buildid), you will be asked 
+    to repeat this string as verification when trying to do install, configure or removal actions.
```
Comment (ned21) [inline quattor/types/aii.pan]: "repeat this string" -> expand this - where do I have to specify it? I think it's the --confirmation argument to aii-shellfe, correct?

**Comment by stdweird** [comment]
@kwaegema make sure this compiles. also squash the commits in one

**Comment by kwaegema** [comment]
@stdweird this compiles in our tree. don't know why the test is not working with this PR..

**Comment by jrha** [comment]
I do, see #122.

**Comment by stdweird** [comment]
retest this please

**Comment by stdweird** [comment]
@jrha builds again

**Comment by jrha** [comment]
Yay!

**Pull request merged**